src/app/components/account-dialog/account-dialog.component.ts

```diff
@@ -0,0 +1,268 @@
+                description: `${formattedAmount} ${symbol}`,
+                icon: 'deposit',
+                userToken: this.token,
+            });
+            const raidenAddress = this.raidenService.raidenAddress;
+            const transferValue = (
+                this.form.value.amount as BigNumber
+            ).toFixed();
+
+            if (this.ethSelected) {
+                return from(
+                    this.web3.eth.sendTransaction({
+                        from: this.defaultAccount,
+                        to: raidenAddress,
+                        value: transferValue,
+                    })
+                );
+            } else {
+                const tokenContract = new this.web3.eth.Contract(
+                    tokenabi,
+                    this.token.address
+                );
+                return from(
+                    tokenContract.methods
+                        .transfer(raidenAddress, transferValue)
+                        .send({ from: this.defaultAccount })
+                );
+            }
+        })
+            .pipe(
+                finalize(() =>
+                    this.notificationService.removePendingAction(
+                        notificationIdentifier
+                    )
+                )
+            )
+            .subscribe({
+                next: () =>
+                    this.notificationService.addSuccessNotification({
+                        title: `Sent ${symbol} to account`,
+                        description: `${formattedAmount} ${symbol}`,
+                        icon: 'deposit',
+                        userToken: this.token,
+                    }),
+                error: (error) =>
+                    this.notificationService.addErrorNotification({
+                        title: `Send ${symbol} to account failed`,
+                        description: error?.message
+                            ? error.message
+                            : error.toString(),
+                        icon: 'error-mark',
+                        userToken: this.token,
+                    }),
+            });
+    }
+
+    private connectWallet() {
+        const requestAccounts$ = defer(() =>
+            from(this.web3.eth.requestAccounts())
+        ).pipe(
+            catchError((error) => {
+                this.accountRequestRejected = true;
+                return throwError(error);
+            })
+        );
+
+        defer(() => {
+            this.requesting = true;
+            this.accountRequestRejected = false;
+            this.wrongChainID = false;
+            return zip(
+                from(this.web3.eth.getChainId()),
+                this.raidenService.network$
+            );
+        })
+            .pipe(
+                switchMap(([web3ChainId, network]) => {
+                    if (web3ChainId !== network.chainId) {
+                        this.wrongChainID = true;
+                        return throwError('Chain ids not matching.');
```

Comment:
```suggestion
                        throw 'Chain ids not matching.';
```
`throwError` receiving the error directly is getting deprected in rxj@7, and specially when inside a `*Map` operator, it's virtually the same as throwing the error directly, so it'd be good to make it slightly more future-proof

Reply:
Thanks for the hint.